### PR TITLE
QA suggestions and Bug Fix

### PR DESCRIPTION
**Pull request by MikeHathaway**
<!---
No need to add special tag
src/ & non src/ changes you need the following (that apply):
-->
# Description of change
## High level
* Various small cleanups to `StandardFunding.sol`
  * Remove `getFundingPowerVotes` due to discovered inconsistencies in `sqrt` calculations in the invariant tests. Moves responsibility for determining votes for a given voting power onto the client.
  * [remove account_ param from _screeningVote](https://github.com/ajna-finance/ecosystem-coordination/pull/76/commits/781d2e6f23853bacc5e5f5ba961863efcaec5b40)
  * [remove unecessary local variable to save on gas in updateSlate](https://github.com/ajna-finance/ecosystem-coordination/pull/76/commits/ca2a1c7cc8130056832d5decd3721ce686cc97ef)
  * [Quanstamp feedback](https://github.com/ajna-finance/ecosystem-coordination/pull/76/commits/2c2b81d8ad8711f2bc838ce8919daa5ca24b9d38)
  * [Fix bug in `startNewDistributionPeriod` causing missed treasury updates](https://github.com/ajna-finance/ecosystem-coordination/pull/76/commits/f8a8f60e734a180d33afbe5e6957febd19ca5fdc)
  * Fix `testMultipleDistributionPeriod`
  * Update associated tests.

# Gas usage

Minor decreases in the gas costs of `propose`, `screeningVote`, and `updateSlate`.

## Pre Change
```
| src/grants/GrantFund.sol:GrantFund contract |                 |        |        |        |         |                                                                                                             
|---------------------------------------------|-----------------|--------|--------|--------|---------|                                                                                                             
| Deployment Cost                             | Deployment Size |        |        |        |         |                                                                                                             
| 3365582                                     | 16817           |        |        |        |         |                                                                                                             
| Function Name                               | min             | avg    | median | max    | # calls |                                                                                                             
| claimDelegateReward                         | 690             | 57029  | 58480  | 68080  | 278     |                                                                                                             
| execute                                     | 8843            | 28549  | 37573  | 47327  | 7       |                                                                                                             
| fundTreasury                                | 8178            | 41716  | 44818  | 65765  | 24      |                                                                                                             
| fundingVote                                 | 1605            | 105784 | 105398 | 416206 | 292     |                                                                                                             
| getDelegateReward                           | 3117            | 3607   | 3117   | 5159   | 5       |                                                                                                             
| getDescriptionHash                          | 906             | 927    | 930    | 930    | 8       |                                                                                                             
| getDistributionId                           | 368             | 418    | 368    | 2368   | 668     |                                                                                                             
| getDistributionPeriodInfo                   | 1066            | 1975   | 1066   | 5066   | 66      |                                                                                                             
| getFundedProposalSlate                      | 1132            | 1273   | 1367   | 1367   | 5       |                                                                                                             
| getFundingVotesCast                         | 1601            | 2341   | 2341   | 3081   | 2       |                                                                                                             
| getProposalInfo                             | 978             | 978    | 978    | 978    | 106     |                                                                                                             
| getSlateHash                                | 889             | 897    | 901    | 901    | 3       |                                                                                                             
| getStage                                    | 1485            | 2880   | 1591   | 9591   | 9       |                                                                                                             
| getTopTenProposals                          | 1165            | 2355   | 2341   | 3282   | 16      |                                                                                                             
| getVoterInfo                                | 1034            | 1034   | 1034   | 1034   | 5       |                                                                                                             
| getVotesFunding                             | 2417            | 6577   | 2825   | 11878  | 15      |                                                                                                             
| getVotesScreening                           | 5201            | 5205   | 5201   | 6107   | 564     |                                                                                                             
| hashProposal                                | 4029            | 4029   | 4029   | 4029   | 52      |                                                                                                             
| propose                                     | 2914            | 75052  | 83723  | 83723  | 64      |                                                                                                             
| screeningVote                               | 1470            | 38504  | 34218  | 399662 | 310     |                                                                                                             
| screeningVotesCast                          | 675             | 1341   | 675    | 2675   | 3       |                                                                                                             
| startNewDistributionPeriod                  | 636             | 34189  | 36258  | 53097  | 23      |                                                                                                             
| state                                       | 1131            | 2021   | 2482   | 2777   | 7       |                                                                                                             
| treasury                                    | 352             | 352    | 352    | 352    | 5       |                                                                                                             
| updateSlate                                 | 1160            | 39978  | 51722  | 78620  | 17      | 
```

## Post Change
```
| src/grants/GrantFund.sol:GrantFund contract |                 |        |        |        |         |                                                                                                             
|---------------------------------------------|-----------------|--------|--------|--------|---------|                                                                                                             
| Deployment Cost                             | Deployment Size |        |        |        |         |                                                                                                             
| 3346358                                     | 16721           |        |        |        |         |                                                                                                             
| Function Name                               | min             | avg    | median | max    | # calls |                                                                                                             
| claimDelegateReward                         | 645             | 53058  | 58435  | 68035  | 75      |                                                                                                             
| execute                                     | 8887            | 32260  | 37617  | 47371  | 10      |                                                                                                             
| fundTreasury                                | 8178            | 41840  | 44818  | 65765  | 25      |                                                                                                             
| fundingVote                                 | 1627            | 105664 | 105420 | 416228 | 94      |                                                                                                             
| getDelegateReward                           | 3094            | 3584   | 3094   | 5136   | 5       |                                                                                                             
| getDescriptionHash                          | 906             | 927    | 930    | 930    | 11      |                                                                                                             
| getDistributionId                           | 368             | 497    | 368    | 2368   | 277     |                                                                                                             
| getDistributionPeriodInfo                   | 1088            | 2035   | 1088   | 5088   | 76      |                                                                                                             
| getFundedProposalSlate                      | 1154            | 1242   | 1154   | 1389   | 8       |                                                                                                             
| getFundingVotesCast                         | 1578            | 2318   | 2318   | 3058   | 2       |                                                                                                             
| getProposalInfo                             | 1000            | 1000   | 1000   | 1000   | 112     |                                                                                                             
| getSlateHash                                | 911             | 919    | 923    | 923    | 3       |                                                                                                             
| getStage                                    | 1485            | 2349   | 1591   | 9591   | 20      |                                                                                                             
| getTopTenProposals                          | 1187            | 2189   | 2363   | 3304   | 19      |                                                                                                             
| getVoterInfo                                | 1034            | 1034   | 1034   | 1034   | 5       |                                                                                                             
| getVotesFunding                             | 2439            | 6599   | 2847   | 11900  | 15      |                                                                                                             
| getVotesScreening                           | 5156            | 5181   | 5156   | 6137   | 161     |                                                                                                             
| hashProposal                                | 4029            | 4029   | 4029   | 4029   | 55      |                                                                                                             
| propose                                     | 2936            | 76273  | 83868  | 83868  | 67      |                                                                                                             
| screeningVote                               | 1902            | 46956  | 34492  | 398507 | 112     |                                                                                                             
| screeningVotesCast                          | 652             | 1318   | 652    | 2652   | 3       |                                                                                                             
| startNewDistributionPeriod                  | 571             | 36512  | 36196  | 52240  | 28      |                                                                                                             
| state                                       | 1065            | 1958   | 2416   | 2711   | 10      |                                                                                                             
| treasury                                    | 374             | 374    | 374    | 374    | 8       |                                                                                                             
| updateSlate                                 | 1204            | 44548  | 70147  | 78725  | 20      |                                                                                                             
                                                                                                                                                                                                                
```

### CODE REVIEW

File: src/grants/base/StandardFunding.sol [hunk 1]
```diff
@@ -983,6 +976,28 @@ abstract contract StandardFunding is Funding, IStandardFunding {
         return keccak256(abi.encode(proposalIds_));
     }
 
+    /**
+        * @notice Retrieve the current stage of the current distribution period.
+        * @return stage_ The stage of the current distribution period.
+     */
+    function getStage() external view returns (string memory stage_) {
```
Comment (prat-gpt) [inline src/grants/base/StandardFunding.sol]: i think this should be added to the `IStandardFunding` interface.

**Review by prat-gpt — APPROVED** [review]
lgtm. added a comment.

File: src/grants/base/StandardFunding.sol (hunk 1 above)
Comment (MikeHathaway) [reply src/grants/base/StandardFunding.sol]: Added

Comment (ith-harvey) [reply src/grants/base/StandardFunding.sol]: Should this also include an "Execution" stage?

File: src/grants/GrantFund.sol [hunk 2]
```diff
@@ -743,9 +740,12 @@ contract GrantFund is IGrantFund, Storage, ReentrancyGuard {
         if (voteCastIndex != -1) {
             // since we are converting from int256 to uint256, we can safely assume that the value will not overflow
             FundingVoteParams storage existingVote = votesCast[uint256(voteCastIndex)];
+            int256 votesUsed = existingVote.votesUsed;
 
             // can't change the direction of a previous vote
-            if (support == 0 && existingVote.votesUsed > 0 || support == 1 && existingVote.votesUsed < 0) {
+            if (
+                (support == 0 && votesUsed > 0) || (support == 1 && votesUsed < 0)
+            ) {
```
Comment (EdNoepel) [inline src/grants/GrantFund.sol:743]: nit: would be more readable on single line

**Review by EdNoepel — APPROVED** [review]
LGTM

**Pull request merged**